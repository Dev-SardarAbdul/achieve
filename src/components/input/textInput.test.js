import { render, fireEvent } from "@testing-library/react"; // Import RTL functions
import TextInput from "./textInput";

it("stores user input value", () => {
  // Render the component
  const { getByPlaceholderText } = render(<TextInput />);

  const inputElement = getByPlaceholderText("Enter your email");

  fireEvent.change(inputElement, { target: { value: "test@example.com" } });

  expect(inputElement.value).toBe("test@example.com");
});
