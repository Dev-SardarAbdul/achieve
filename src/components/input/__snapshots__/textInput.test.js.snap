// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<div
  className="MuiBox-root css-0"
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <div
    className="MuiFormControl-root MuiTextField-root css-oqo43x-MuiFormControl-root-MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Enter your email"
        required={false}
        type="email"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-ihdtdm"
        >
          <span
            className="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
